spring-rabbit-stream/src/test/java/org/springframework/rabbit/stream/listener/AbstractIntegrationTests.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.rabbit.stream.listener;
+
+import java.time.Duration;
+
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.utility.DockerImageName;
+
+/**
+ * @author Gary Russell
+ * @since 2.4
+ *
+ */
+public abstract class AbstractIntegrationTests {
+
+	static final GenericContainer<?> rabbitmq = new GenericContainer<>(
+			DockerImageName.parse("pivotalrabbitmq/rabbitmq-stream"))
+				.withExposedPorts(5672, 15672, 5552)
+				.withStartupTimeout(Duration.ofMinutes(2));
+
+	static {
+		rabbitmq.start();
```

Comment:
No; `TestContainers` takes care of it

>2021-06-16 14:08:17,379  INFO org.testcontainers.DockerClientFactory [main] : Ryuk started - will monitor and terminate Testcontainers containers on JVM exit

Reply:
> JVM exit

But we may run the whole test suite, so all those container are going to live until our Gradle build is done.
Correct me if I'm wrong.
Thanks

Reply:
I don't see any choice right now; I don't want to start a new container for each test class; hence the abstract class. I don't think it will hurt much to have an idle docker container.